python/paddle/nn/layer/container.py

```diff
@@ -319,6 +319,124 @@ def update(
                 self.add_sublayer(kv[0], kv[1])
 
 
+class ParameterDict(Layer):
+    """
+    Holds parameters in a dictionary.
+
+    ParameterDict can be indexed like a regular Python dictionary, but Parameters it contains are properly registered.
+
+    Parameters:
+        values (iterable, optional): a mapping (dictionary) of (string : Any) or an iterable of key-value pairs of type (string, Any)
+
+    Examples:
+        .. code-block:: python
+
+            >>> import paddle
+
+            >>> class MyLayer(paddle.nn.Layer):
+            ...     def __init__(self, num_stacked_param):
+            ...         super().__init__()
+            ...         # create ParameterDict with iterable Parameters
+            ...         self.params = paddle.nn.ParameterDict(
+            ...             {'t' + str(i): paddle.create_parameter(shape=[2, 2], dtype='float32') for i in range(num_stacked_param)})
+            ...
+            ...     def forward(self, x):
+            ...         for i, (key, p) in enumerate(self.params):
+            ...             tmp = self._helper.create_variable_for_type_inference('float32')
+            ...             self._helper.append_op(
+            ...                 type="mul",
+            ...                 inputs={"X": x,
+            ...                         "Y": p},
+            ...                 outputs={"Out": tmp},
+            ...                 attrs={"x_num_col_dims": 1,
+            ...                         "y_num_col_dims": 1})
+            ...             x = tmp
```

Comment:
please use public APIs instead of non-public APIs(`self._helper.create_variable_for_type_inference` and `self._helper.append_op`) in the example code. for example, `x = paddle.multiply(x, p)` or `x = x * p`